### DISCUSSION TITLE
Get Current Tenant from Service Provider

### DISCUSSION THREAD

**Discussion by seanlabs** [spatie/laravel-multitenancy#485] [Q&A]
Is there a way to get the current tenant in a custom service provider? I am actually using spatie's little gatekeeper and I would like each tenant to be able to have their own username/password instead of defining a single password in the .env file that spans multiple tenants. 

I've created a custom service provider that extends the little gatekeeper service provider, but when I try to get the current tenant with app('currentTenant') or using Tenant::current() it throws errors. 

How would you access the tenant in a custom service provider or any service provider for that matter?

**Comment by masterix21** [comment]
I don't use the [spatie/laravel-littlegatekeeper](https://github.com/spatie/laravel-littlegatekeeper), but I think that creating a custom ServiceProvider that replaces the original one (`LittleGateKeeperServiceProvider.php`), you can reach your goal. 

@freekmurze or @sebastiandedeyne, exists a better approach?

**Reply by seanlabs** [reply]
@masterix21 - I tried that, but it throws error when trying to access app('currentTenant'). I think its not loaded yet with the laravel app stack, so its not available. Anybody have any ideas? @freekmurze @sebastiandedeyne

**Reply by masterix21** [reply]
It could be a priority issue. Try registering the multitenancy service provider in the `config/app.php` before the little gatekeeper.

**Reply by danielbehrendt** [reply]
I'm interested on how I can have `app('currentTenant')` available in other service providers. I already tried to disable auto-discovery via composer.json and registering them in the required order in `config/app.php` but it seems not to work because the current tenant is still `null`. Any ideas @freekmurze and @sebastiandedeyne?

**Reply by masterix21** [reply]
Please create a public repository to test the issue.

**Reply by danielbehrendt** [reply]
@masterix21 i created a test repo and noticed that `app('currentTenant')` is available in the `boot` method of my custom service provide but not in the `register` method. the problem here is that i have a service provider of a community package which does some heavy lifting in the `register` method and i need to extend/overwrite this by setting some tenant related config before that.

**Reply by danielbehrendt** [reply]
@masterix21 i narrowed the problem down to the place where `app('currentTenant')` isn't available in the `TranslationServiceProvider`.

**Reply by seanlabs** [reply]
@danielbehrendt

have you found any new information? I'm still stuck.

**Reply by nocodelab** [reply]
Any update on this one?
